client/login/wp-login/index.jsx

```diff
@@ -575,39 +595,82 @@ export class Login extends Component {
 			</svg>
 		);
 
+		const mainContent = (
+			<Main
+				className={ clsx( 'wp-login__main', {
+					'is-wpcom-migration': isFromMigrationPlugin,
+					'is-social-first': isSocialFirst,
+					'is-generic-oauth': isGenericOauth,
+				} ) }
+			>
+				{ this.renderI18nSuggestions() }
+
+				<DocumentHead
+					title={ translate( 'Log In' ) }
+					link={ [ { rel: 'canonical', href: canonicalUrl } ] }
+					meta={ [
+						{
+							name: 'description',
+							content: translate(
+								'Log in to your WordPress.com account to manage your website, publish content, and access all your tools securely and easily.'
+							),
+						},
+					] }
+				/>
+
+				<div className="wp-login__container">{ this.renderContent( isSocialFirst ) }</div>
+			</Main>
+		);
+
+		const headerText = (
+			<GetHeaderText
+				isSocialFirst={ isSocialFirst }
+				twoFactorAuthType={ twoFactorAuthType }
+				isManualRenewalImmediateLoginAttempt={ isManualRenewalImmediateLoginAttempt }
+				socialConnect={ socialConnect }
+				linkingSocialService={ linkingSocialService }
+				action={ action }
+				privateSite={ privateSite }
+				oauth2Client={ oauth2Client }
+				isWooJPC={ isWooJPC }
+				isFromMigrationPlugin={ isFromMigrationPlugin }
+				isJetpack={ isJetpack }
+				isWCCOM={ isWCCOM }
+				isFromAkismet={ isFromAkismet }
+				isFromAutomatticForAgenciesPlugin={ isFromAutomatticForAgenciesPlugin }
+				isGravPoweredClient={ isGravPoweredClient }
+				wccomFrom={ wccomFrom }
+				twoFactorEnabled={ twoFactorEnabled }
+				currentQuery={ currentQuery }
+				showContinueAsUser={ showContinueAsUser }
+			/>
+		);
+
+		let brandLogo;
+
+		if ( isFromAkismet ) {
+			brandLogo = akismetLogo;
+		} else if (
+			isPartnerPortalOAuth2Client( oauth2Client ) &&
+			document.location.search?.includes( 'wpcloud' )
+		) {
+			brandLogo = <WPCloudLogo className="login__wpcloud-logo" size={ 120 } />;
+		}
+
 		return (
 			<>
 				{ isSocialFirst && (
-					<Step.TopBar
-						rightElement={ this.renderLoginHeaderNavigation() }
-						logo={ isFromAkismet && akismetLogo }
-					/>
+					<Step.CenteredColumnLayout
+						columnWidth={ 6 }
+						topBar={
+							<Step.TopBar rightElement={ this.renderLoginHeaderNavigation() } logo={ brandLogo } />
+						}
+						heading={ <Step.Heading text={ headerText } /> }
+					>
+						{ mainContent }
```

Comment:
We could further break down this mainContent to use StepContainer wrappers but that will require taking apart more bits of LoginBlock. It might be better left to a follow-up PR because this diff is already pretty big.

(I'm not 100% sure we _need_ to have a LoginBlock component, because they're both looking at similar conditions to render things. There's no real separation of concerns.)